src/Interpreters/Cache/FileCache.cpp

```diff
@@ -1136,7 +1142,15 @@ void FileCache::loadMetadata()
             "Please, check log for error messages");
     }
 
-    loadMetadataImpl();
+    if (load_metadata_asynchronously)
+    {
+        ThreadFromGlobalPool load_metadata_main_thread([this] { loadMetadataImpl(); });
+        load_metadata_main_thread.detach();
```

Comment:
We should be able to deactivate this thread in `FileCache::deactivateBackgroundOperations`.